### PR TITLE
Enhance the logging when encounter lock issue

### PR DESCRIPTION
**Pull request by JaySon-Huang**
some logging is not correctly format to output

```
tiflash-1/server.log:[2024/12/03 08:26:39.803 +08:00] [ERROR] [<unknown>] ["coprocessor meets error, error message: {}, error code: {}"] [source="MPP<gather_id:1, query_ts:1733185578514561761, local_query_id:23880, server_id:1855, start_ts:454344200276410379,task_id:1> pingcap/coprocessor"] [thread_id=3853]
tiflash-1/server.log:[2024/12/03 08:26:39.804 +08:00] [INFO] [<unknown>] ["cop task exit because already meet error"] [source="MPP<gather_id:1, query_ts:1733185578514561761, local_query_id:23880, server_id:1855, start_ts:454344200276410379,task_id:1> pingcap/coprocessor"] [thread_id=3853]
tiflash-1/server.log:[2024/12/03 08:26:39.804 +08:00] [WARN] [CoprocessorReaderSourceOp.cpp:78] ["coprocessor reader meets error: primary_lock: \"t\\200\\000\\000\\000\\000\\000\\000r_r\\003\\200\\000\\000\\000\\000\\000\\000P\\003\\200\\000\\000\\000\\000\\000\\000\\003\"\nlock_version: 454344200119123975\nkey: \"t\\200\\000\\000\\000\\000\\000\\000|_r\\003\\200\\000\\000\\000\\000\\000\\000P\\003\\200\\000\\000\\000\\000\\000\\000\\003\\003\\200\\000\\000\\000\\000\\000\\017\\365\\003\\200\\000\\000\\000\\000\\000\\000\\001\"\nlock_ttl: 20041\ntxn_size: 6\nlock_for_update_ts: 454344200119123975\nuse_async_commit: true\nmin_commit_ts: 454344200132231183\n"] [source="MPP<gather_id:1, query_ts:1733185578514561761, local_query_id:23880, server_id:1855, start_ts:454344200276410379,task_id:1>"] [thread_id=13]
tiflash-1/server.log:[2024/12/03 08:26:39.805 +08:00] [WARN] [Task.cpp:128] ["error occurred and cancel the query"] [source="MPP<gather_id:1, query_ts:1733185578514561761, local_query_id:23880, server_id:1855, start_ts:454344200276410379,task_id:1> 1"] [thread_id=13]
```

### CODE REVIEW

File: src/coprocessor/Client.cc [hunk 1]
```diff
@@ -726,10 +727,10 @@ void ResponseIter::handleTask(const CopTask & task)
     {
         if (is_cancelled || meet_error)
             return;
+        const auto & current_task = remain_tasks[idx];
+        auto & bo = bo_maps.try_emplace(current_task.region_id.id, kv::copNextMaxBackoff).first->second;
```
Comment (zanmato1984) [inline src/coprocessor/Client.cc:731]: Is this line of change related to the log format issue?

Comment (JaySon-Huang) [reply src/coprocessor/Client.cc:733]: I found it is hard to locating which region meet the lock error, and which lock_version is prevent the request being handle.
So I move these two lines out of the `try ... catch ...` scope, in order to log down the region_id when error happens in line 744.

Comment (zanmato1984) [reply src/coprocessor/Client.cc:733]: I see, thank you.

**Review by zanmato1984 — APPROVED** [review]

**Comment by JaySon-Huang** [comment]
PTAL @gengliqi

**Review by gengliqi — APPROVED** [review]

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Pull request merged**